src/Stratis.Bitcoin.Features.SmartContracts/Networks/SmartContractNetwork.cs

```diff
@@ -0,0 +1,46 @@
+﻿using NBitcoin;
+using NBitcoin.DataEncoders;
+using Stratis.Bitcoin.Features.SmartContracts.Consensus;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.Networks
+{
+    public static class SmartContractNetwork
+    {
+        public static Block CreateGenesis(ConsensusFactory consensusFactory, uint genesisTime, uint nonce, uint bits, int version, Money reward)
```

Comment:
Building the transaction and the block could be is seperate methods.  Not too sure if this adds anything.  However, if the calling code wants to see the genesis txn for any reason, then `CreateGenesis()` could return a tuple for `Block` and `Transaction`.  

If there is no need for this right now don't worry, just an idea.